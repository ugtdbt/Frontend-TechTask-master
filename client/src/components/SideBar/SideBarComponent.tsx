import React, { memo } from "react";
import type { FC } from "react";
import Grid from "@mui/material/Grid";
import Typography from "@mui/material/Typography";
import Link from "@mui/material/Link";

interface SideBarProps {
  list: ReadonlyArray<{
    name: string;
    urlPath: string;
  }>;
}

const SideBar: FC<SideBarProps> = memo((props: SideBarProps) => {
  const { list } = props;

  if (list.length === 0) {
    return <>Loading ....</>;
  }

  return (
    <Grid item xs={12} md={2} data-testid="sidebar">
      <Typography variant="h6" pl={3} gutterBottom sx={{ mt: 0 }}>
        Kategorien
      </Typography>
      {list.map((archive: any) => (
        <Link
          display="block"
          variant="body1"
          pl={3}
          pt={1}
          href={archive.urlPath}
          key={archive.name}
          className="sideBar-item"
        >
          {archive.name}
        </Link>
      ))}
    </Grid>
  );
});

SideBar.displayName = "SideBar";
export default SideBar;
