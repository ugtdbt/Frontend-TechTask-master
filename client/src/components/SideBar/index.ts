export { default as SideBar } from "./SideBarComponent";
