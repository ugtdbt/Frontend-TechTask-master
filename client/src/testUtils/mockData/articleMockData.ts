export const article = {
  imageUrl:
    "https://cdn1.home24.net/images/media/catalog/product/200x200/png/l/a/lattenrost-super-fix-90-x-200cm-4448116.webp",
  name: "Lattenrost Super Fix",
  variantName: "Breite: 140 cm",
  price: 14990,
};
