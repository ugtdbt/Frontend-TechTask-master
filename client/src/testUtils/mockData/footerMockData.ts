export const footer = {
  description:
    "Alle Preise sind in Euro (â‚¬) inkl. gesetzlicher Umsatzsteuer und Versandkosten.",
};
