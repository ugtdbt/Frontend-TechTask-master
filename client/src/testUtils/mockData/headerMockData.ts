export const sections = [
  { title: "furniture", url: "#" },
  { title: "lamps", url: "#" },
  { title: "accessories", url: "#" },
  { title: "textiles", url: "#" },
  { title: "children", url: "#" },
  { title: "kitchen", url: "#" },
  { title: "bath", url: "#" },
  { title: "Garden", url: "#" },
  { title: "Brands", url: "#" },
  { title: "Shop the look", url: "#" },
  { title: "advisory", url: "#" },
  { title: "stores", url: "#" },
  { title: "Sale", url: "#" },
];

export const header = {
  sections,
  title: "Home24",
};
