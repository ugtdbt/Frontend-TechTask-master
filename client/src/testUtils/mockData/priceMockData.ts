export const price = {
  price: 2500,
};
