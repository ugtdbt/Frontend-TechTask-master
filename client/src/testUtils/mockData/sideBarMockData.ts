export const sideBar = {
  list: [
    { name: "Wohnzimmer", urlPath: "wohnzimmer-moebel/" },
    { name: "Polsterm√∂bel", urlPath: "alle-polstermoebel/" },
  ],
};
