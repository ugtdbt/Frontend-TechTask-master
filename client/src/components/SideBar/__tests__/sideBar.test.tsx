import { cleanup, render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import { sideBar } from "../../../testUtils/mockData/sideBarMockData";
import SideBar from "../SideBarComponent";

describe("SideBar component", () => {
  const SideBarComponent = <SideBar list={sideBar.list} />;

  afterEach(() => {
    cleanup();
  });

  it("SideBar Match Snapshot", () => {
    const tree = renderer.create(SideBarComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render SideBar component", () => {
    render(SideBarComponent);
    const sideBarElem = screen.getByTestId("sidebar");
    expect(sideBarElem).toBeInTheDocument();
  });

  it("Render the 1st menu as Kategorien", () => {
    render(SideBarComponent);
    const _MenuItems = document.querySelectorAll(".sideBar-item");
    expect(_MenuItems[0].innerHTML).toContain(sideBar.list[0].name);
  });

  it("Render the 2nd menu as Schlafzimmer", () => {
    render(SideBarComponent);
    const _MenuItems = document.querySelectorAll(".sideBar-item");
    expect(_MenuItems[1].innerHTML).toContain(sideBar.list[1].name);
  });
});
