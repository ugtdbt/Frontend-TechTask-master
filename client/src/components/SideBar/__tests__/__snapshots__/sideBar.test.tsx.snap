// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component SideBar Match Snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 css-13i4rnv-MuiGrid-root"
  data-testid="sidebar"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-15zetcd-MuiTypography-root"
  >
    Kategorien
  </h6>
  <a
    className="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways sideBar-item css-1qlkw6l-MuiTypography-root-MuiLink-root"
    href="wohnzimmer-moebel/"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Wohnzimmer
  </a>
  <a
    className="MuiTypography-root MuiTypography-body1 MuiLink-root MuiLink-underlineAlways sideBar-item css-1qlkw6l-MuiTypography-root-MuiLink-root"
    href="alle-polstermoebel/"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Polsterm√∂bel
  </a>
</div>
`;
